### PR TITLE
Refactored ResourceManager Docker support

### PR DESCRIPTION
**Pull request by bgulla**
Highlights:
  - Hadoop distribution is now user definable. Use the HADOOP_TARBALL_URL environment variable to force the docker to download and use your distribution.
  - Hadoop 2.7.1 is now the default distribution used.
  - The base image is now based off of Debian for leaner docker images.
  - ResourceManager now runs as 'yarn' user.

### CODE REVIEW

File: myriad-scheduler/src/main/resources/myriad-config-default.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
-mesosMaster: 10.0.2.15:5050
+mesosMaster: 172.31.45.229:5050
 checkpoint: false
 frameworkFailoverTimeout: 43200000
-frameworkName: MyriadAlpha
+frameworkName: Myriad
```
Comment (smarella) [inline myriad-scheduler/src/main/resources/myriad-config-default.yml]: @bgulla Looks like these config changes were made to match your specific environment. Is that correct? If yes, can you please remove this file from the PR.

**Comment by smarella** [comment]
I skimmed through the changes and they look good to me. But I'm not a docker expert to comment on some of the details. Any volunteers to take a deeper look?

@bgulla Is this something that you'd like to have submitted for 0.1 release? If yes, can you please mark the corresponding JIRA with "fix version" set to myriad-0.1.0 ?

**Comment by bgulla** [comment]
I reverted the myriad-config-default.yml file, looked like my local dev setup file snuck its way into the commit.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#
+# VERSION 0.0.1
+
+FROM debian
+MAINTAINER Apache Myriad dev@myriad.incubator.apache.org
+#USER yarn
+
+ENV HADOOP_USER="yarn"
+ENV HADOOP_VER="2.7.1"
+ENV HADOOP_TARBALL_URL="http://172.31.1.11/hadoop-2.7.1.tar.gz"
```
Comment (sarjeet2013) [inline docker/Dockerfile]: Replace this IP as well? Here and other places wherever it exists.

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+myriad:
+  build: .
+#    - dockerfile: ResourceManager.dockerfile
```
Comment (sarjeet2013) [inline docker/docker-compose.yml]: Would you need to uncomment this as you've deleted "Dockerfile" file from last diff, and now only require "ResourceManager.dockerfile" file to build RM docker image?

And, A bit curious to know how docker-compose.yml works here?

Comment (bgulla) [inline docker/docker-compose.yml]: You're absolutely right. I have commited the fix.

**Comment by sarjeet2013** [comment]
LGTM.

**Pull request closed without merging**